### PR TITLE
:sparkles: Enable secureboot for all flavors (minus alpine)

### PR DESCRIPTION
**Pull request by Itxaka**
This PR brings new deps, bumps builders and agent to bring secureboot in all flavors but alpine (more ont hat below)

Currently to enable secureboot in a given X flavor we need to make sure to use the artifacts from that falvor specifically as they are all signed with the distro owner key, so the chain of validation works as expected.

shim (dual signed Microsoft and distro owner) -> grub (distro owner signed) -> kernel (distro owner signed)

So unless there comes a time where the shim is signed by Microsoft and ALL the distro owners (so you can chainload all grubs and kernel signed by all of them) we need to specifically use the distro artifacts so the signatures match.

This PR brings a new enki version(inside osbuilder) which builds the iso using the specifci distro artifacts (instead of generic ones as we used to do).

It also makes sure that all distros include the grub and shim signed artifacts in the base images.

Also brings a new framework that drops the generic grub artifacts (not used) and it also brings a new agent that uses those new artifacts during installation instead of the generic ones.

Unfortunately Alpine does not provide signed artifacts so SecureBoot is not supported under Alpine until they provide those. We still use our generic artifacts to enable uefi support for Alpine

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by Itxaka** [comment]
leap 15.5:
 - boots with secureboot
 - installs with secureboot
 - no regexp module in the grub directly so cmdline is NOT correct. we need to ship the module during install and insmod it

Fedora 38:
 - boots with secureboot
 - installs with secureboot
 - regexp modules is bundled in grub

**Comment by Itxaka** [comment]
Debian bookworm:
 - boots with secureboot
 - installs with secureboot
 - regexp bundled with grub


Ubuntu 23.10(probably the rest):
 - NOT boot with secureboot(incorrect naming, grub is stored as grubx64.efi but its looking for grub.efi)
image

**Comment by Itxaka** [comment]
looks like on ubuntu is failing to find the actual shim and using the fallback so that is why it cant find the proper named thingie. So it may be a package missing

```
                +iso | INFO[2024-01-12T08:51:42Z] Creating EFI image...                        
                +iso | DEBU[2024-01-12T08:51:42Z] skip copying /tmp/enki-iso683099797/rootfs/usr/share/efi/x86_64/shim.efi: not found 
                +iso | DEBU[2024-01-12T08:51:42Z] skip copying /tmp/enki-iso683099797/rootfs/usr/lib/shim/shimx64.efi.signed: not found 
                +iso | DEBU[2024-01-12T08:51:42Z] skip copying /tmp/enki-iso683099797/rootfs/boot/efi/EFI/fedora/shim.efi: not found 
                +iso | DEBU[2024-01-12T08:51:42Z] skip copying /tmp/enki-iso683099797/rootfs/boot/efi/EFI/rocky/shim.efi: not found 
                +iso | DEBU[2024-01-12T08:51:42Z] skip copying /tmp/enki-iso683099797/rootfs/boot/efi/EFI/redhat/shim.efi: not found 
                +iso | DEBU[2024-01-12T08:51:42Z] skip copying /tmp/enki-iso683099797/rootfs/boot/efi/EFI/almalinux/shim.efi: not found 
                +iso | DEBU[2024-01-12T08:51:42Z] Using fallback shim file /efi/EFI/BOOT/bootx64.efi 
```

**Comment by Itxaka** [comment]
yeah and the root problem is:

 - shimx64.signed is a link to the alternatives as you can have newer shims `lrwxrwxrwx  1 root root     36 Apr 16  2020 shimx64.efi.signed -> /etc/alternatives/shimx64.efi.signed`
 - in our rootfs alternatives is not installed so you get an empty link
 - testing in a normal image properly links the file


So the solution is to try and pick the `shimx64.efi.signed.latest` file instead or even better, they have a dualsigned one `shimx64.efi.dualsigned` signed by them AND microsoft, so thats even better.

**Comment by Itxaka** [comment]
Using fix https://github.com/kairos-io/enki/pull/25 resutls in proper isos with secureboot for all distros (minus alpine)
Installs correctly for all distros
leap/opensuse still missing the regexp module bundled, will need to check further

**Comment by Itxaka** [comment]
with that merged and pacakge bumped, the only missing fix is for agent:

 - On ubuntu install copy the `grub.cfg` under `/EFI/ubungu/grub.cfg`
 - On opensuse install, copy the regexp module for grub (check the proper paths where its copied)


The default grub config already does an `insmod regexp` so either is not being copied properly or its missing from the sources or its just on a wrong path.

File: Earthfile [hunk 1]
```diff
@@ -207,12 +207,20 @@ luet:
 kairos-dockerfile:
     ARG --required FAMILY
     COPY ./images .
-    RUN --no-cache cat <(echo "# This file is auto-generated with the command: earthly +kairos-dockerfile --FAMILY=${FAMILY}") \
-        <(sed -n '/# WARNING:/!p' Dockerfile.$FAMILY) \
-        <(echo) \
-        <(sed -n '/# WARNING:/!p' Dockerfile.kairos) \
-        > ./Dockerfile
-    SAVE ARTIFACT Dockerfile AS LOCAL images/Dockerfile.kairos-${FAMILY}
+    IF [ "$FAMILY" == "all" ]
```
Comment (Itxaka) [inline Earthfile:210]: ehm, this is something that I added to be able to generate all flavors dockerfiles at the same time as I was getting upset doing it one by one.

Can be dropped

**Comment by Itxaka** [comment]
https://github.com/kairos-io/kairos-agent/pull/203 fixes ubuntu boot.

File: Earthfile (hunk 1 above)
Comment (jimmykarily) [reply Earthfile:210]: nice! I wanted to do the same but didn't get the chance.

File: Earthfile [hunk 2]
```diff
@@ -286,6 +294,14 @@ base-image:
     ARG _CIMG=$(cat ./IMAGE)
 
     COPY +git-version/GIT_VERSION VERSION
+    ARG KAIROS_AGENT_DEV_BRANCH
+
+    IF [ "$KAIROS_AGENT_DEV_BRANCH" != "" ]
```
[the remark this answers is not in the document]
Comment (jimmykarily) [reply Earthfile:299]: good idea :+1:

**Review by mauromorales — APPROVED** [review]
nice!

**Comment by Itxaka** [comment]
latest fixes should fix:
 - ubuntu grub copy https://github.com/kairos-io/kairos-agent/pull/203
 - opensuse regexp modules missing  https://github.com/kairos-io/packages/pull/619

**Comment by Itxaka** [comment]
both fixes deployed via framework 2.6.3, needs manual QA

File: tests/tests_suite_test.go [hunk 3]
```diff
@@ -304,6 +304,19 @@ func expectRebootedToActive(vm VM) {
...
 }
 
+func expectSecureBootEnabled(vm VM) {
+	// Check for secureboot before install, based on firmware env var
+	// if we set, then the test suite will load the secureboot firmware
+	secureboot := os.Getenv("FIRMWARE")
+
+	if secureboot == "true" {
+		By("checking that secureboot is enabled", func() {
+			out, _ := vm.Sudo("dmesge | grep -i secure")
```
Comment (jimmykarily) [inline tests/tests_suite_test.go:314]: `dmesge` -> `dmesg` ? This shouldn't work...

Comment (Itxaka) [reply tests/tests_suite_test.go:314]: Manually testing this, very weird indeed because the test actually passed??

Comment (Itxaka) [reply tests/tests_suite_test.go:314]: Ah crap, maybe the if it's not correct so it doesn't go over this...

Comment (Itxaka) [reply tests/tests_suite_test.go:314]: Yes the check is wrong, I'm stupid

Comment (Itxaka) [reply tests/tests_suite_test.go]: fixed, good catch @jimmykarily !!

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**